src/lightbox.ts

```diff
@@ -0,0 +1,73 @@
+import { html, component } from '@pionjs/pion'
+import styles from './lightbox.css'
+import type { Viewable, Props } from './types'
+import useLightbox from './hooks/useLightbox'
+import { clearIcon } from './components/icons'
+
+const Lightbox = (props: Props) => {
+  const { slide, close } = useLightbox(props)
+
+  return html`
+    <style>
+      ${styles}
+    </style>
+
+    <cosmoz-slider .slide=${slide}></cosmoz-slider>
+
+    <button class="close" @click=${close} aria-label="Close lightbox">
+      ${clearIcon()}
+    </button>
+  `
+}
+
+// Define the custom element
+customElements.define('pion-lightbox', component<Props>(Lightbox))
```

Comment:
You should use the (not yet documented) `styleSheets` option to provide a component stylesheet. Internally it uses `adoptedStyleSheets` to efficiently re-use the same stylesheet for all component instances.

```suggestion
  return html`
    <cosmoz-slider .slide=${slide}></cosmoz-slider>

    <button class="close" @click=${close} aria-label="Close lightbox">
      ${clearIcon()}
    </button>
  `
}

// Define the custom element
customElements.define('pion-lightbox', component<Props>(Lightbox, {styleSheets: [styles]}))
```